test/system_SUITE.erl

```diff
@@ -0,0 +1,449 @@
+    {ok, Con} =
+     amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
+                                                port = Port,
+                                                ssl_options = [{cert, CertFirst},
+                                                               {key, KeyFirst} ]}),
+    %% verify the other certificate is not accepted
+    {error, ?SERVER_REJECT_CLIENT} =
+     amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
+                                                port = Port,
+                                                ssl_options = [{cert, CertUpdated},
+                                                               {key, KeyUpdated} ]}),
+    ok = amqp_connection:close(Con),
+
+    %% When: a whitelisted certicate is replaced with one with the same name 
+    ok = whitelist(friendlies(), "bart", CertUpdated,  KeyUpdated),
+
+    wait_for_trust_store_refresh(),
+
+    %% Then: the first certificate should be rejected
+    {error, ?SERVER_REJECT_CLIENT} =
+     amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
+                                                port = Port,
+                                                ssl_options = [{cert, CertFirst},
+                                                               %% disable ssl session caching
+                                                               %% as this ensures the cert
+                                                               %% will be re-verified by the
+                                                               %% server
+                                                               {reuse_sessions, false},
+                                                               {key, KeyFirst} ]}),
+
+    %% And: the updated certificate should allow the user to connect
+    {ok, Con2} =
+     amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
+                                                port = Port,
+                                                ssl_options = [{cert, CertUpdated},
+                                                               {reuse_sessions, false},
+                                                               {key, KeyUpdated} ]}),
+    ok = amqp_connection:close(Con2),
+    %% Clean: server TLS/TCP.
+    ok = delete("bart.pem"),
+    ok = rabbit_networking:stop_tcp_listener(Port).
+
+
+ensure_configuration_using_binary_strings_is_handled(Config) ->
+    ok = rabbit_ct_broker_helpers:rpc(Config, 0,
+           ?MODULE, ensure_configuration_using_binary_strings_is_handled1, []). 
+
+ensure_configuration_using_binary_strings_is_handled1() ->
+    ok = change_configuration(rabbitmq_trust_store, [{directory, list_to_binary(friendlies())},
+                                                    {refresh_interval,
+                                                        {seconds, interval()}}]).
+
+%% Test Constants
+
+port() -> 4096.
```

Comment:
Instead of hardcoding a TCP port number, we should give the possibility to testsuites to ask `rabbit_ct_broker_helpers` to compute one for them.

In the meantime, you can:
* add a name for it in `rabbit_common/src/rabbit_ct_broker_helpers.erl` in `TCP_PORTS_LIST`
* add a no-op clause to `update_tcp_ports_in_rmq_config()`